api-registration/server/index.js

```diff
@@ -0,0 +1,63 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  /* your code starts here */
+  argon2
+    .hash(password)
+    .then(hashedPassword => {
+      const sql = `
+      insert into "users" ("username, "hashedPassword")
+      values ($1, $2)
+      returning *`;
+
+      const params = [username, hashedPassword];
+      db.query(sql, params)
+        .then(result => {
+          const user = result.rows[0];
+          res.status(201).json(user);
```

Comment:
This will include the `hashedPassword` in the data sent back to the client, we only want to send the following data to the client:
```JAVASCRIPT
{
    "createdAt": "2020-12-14T16:38:52.995Z",
    "userId": 2,
    "username": "admin"
}
```

You can update the `returning` statement in your SQL to only return the specific columns you want.

Reply:
![Capture](https://user-images.githubusercontent.com/96793078/162275529-283b34f3-5d78-4ee1-b9c5-cd8b98289f49.PNG)
![Capture](https://user-images.githubusercontent.com/96793078/162275613-e5bc7d49-7006-45cd-a736-9eadaaf5edfb.PNG)